### PR TITLE
New constructors for container classes

### LINKED ISSUE #155: Add container constructors that allow passing a list of entities
Currently, passing in a list of entities to constructor container classes like `RawGridElements`, `SystemParticipants` etc. is not supported. It would simplify to provide such constructors because this would simplify container class creation on update. In particular with the existing methods `allEntitesAsList(...)` I think this makes a lot of sense.

### PR DESCRIPTION
**Pull request by johanneshiry**
Resolves #155

### CODE REVIEW

File: src/main/java/edu/ie3/datamodel/models/input/container/GraphicElements.java [hunk 1]
```diff
@@ -52,6 +53,44 @@ public GraphicElements(Collection<GraphicElements> graphicElements) {
...
+        systemParticipantInput -> {
+          if (systemParticipantInput instanceof NodeGraphicInput) {
+            nodeGraphics.add((NodeGraphicInput) systemParticipantInput);
+          } else if (systemParticipantInput instanceof LineGraphicInput) {
+            lineGraphics.add((LineGraphicInput) systemParticipantInput);
+          } else {
+            throw new InvalidEntityException(
+                "Provided entity is not included in GraphicElements.", systemParticipantInput);
+          }
+        });
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/models/input/container/GraphicElements.java:79]:
```suggestion
    this.nodeGraphics =
        graphics
            .parallelStream()
            .filter(graphic -> graphic instanceof NodeGraphicInput)
            .map(graphic -> (NodeGraphicInput) graphic)
            .collect(Collectors.toSet());
    this.lineGraphics =
        graphics
            .parallelStream()
            .filter(graphic -> graphic instanceof LineGraphicInput)
            .map(graphic -> (LineGraphicInput) graphic)
            .collect(Collectors.toSet());
```

Why not this way?

File: src/main/java/edu/ie3/datamodel/models/input/container/GraphicElements.java [hunk 2]
```diff
@@ -52,6 +53,44 @@ public GraphicElements(Collection<GraphicElements> graphicElements) {
...
+          "The provided entities in '"
+              + this.getClass().getSimpleName()
+              + "' contains duplicate UUIDs. "
+              + "This is not allowed!\nDuplicated uuids:\n\n"
+              + exceptionString);
+    }
+  }
+
   @Override
   public List<GraphicInput> allEntitiesAsList() {
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/models/input/container/GraphicElements.java:95]:
```suggestion
  public final List<GraphicInput> allEntitiesAsList() {
```

Might resolve the SonarQube issue!?

File: src/main/java/edu/ie3/datamodel/models/input/container/RawGridElements.java [hunk 3]
```diff
@@ -88,6 +90,56 @@ public RawGridElements(Collection<RawGridElements> rawGridElements) {
...
+  /**
+   * Create an instance based on a list of {@link AssetInput} entities that are included in {@link
+   * RawGridElements}
+   *
+   * @param rawGridElements list of grid elements this container instance should created from
+   */
+  public RawGridElements(List<AssetInput> rawGridElements) {
+
+    /* init sets */
+    this.nodes = new HashSet<>();
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/models/input/container/RawGridElements.java:102]: See suggestion above.

File: src/main/java/edu/ie3/datamodel/models/input/container/RawGridElements.java [hunk 4]
```diff
@@ -88,6 +90,56 @@ public RawGridElements(Collection<RawGridElements> rawGridElements) {
...
+          "The provided entities in '"
+              + this.getClass().getSimpleName()
+              + "' contains duplicate UUIDs. "
+              + "This is not allowed!\nDuplicated uuids:\n\n"
+              + exceptionString);
+    }
+  }
+
   @Override
   public List<AssetInput> allEntitiesAsList() {
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/models/input/container/RawGridElements.java:144]:
```suggestion
  public final List<AssetInput> allEntitiesAsList() {
```

Might as well resolve the SonarQube issue?!

File: src/main/java/edu/ie3/datamodel/models/input/container/SystemParticipants.java [hunk 5]
```diff
@@ -110,6 +111,68 @@ public SystemParticipants(Collection<SystemParticipants> systemParticipants) {
...
+  /**
+   * Create an instance based on a list of {@link SystemParticipantInput} entities
+   *
+   * @param systemParticipants list of system participants this container instance should created
+   *     from
+   */
+  public SystemParticipants(List<SystemParticipantInput> systemParticipants) {
+
+    /* init sets */
+    this.bmPlants = new HashSet<>();
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/models/input/container/SystemParticipants.java:123]: See suggestion above.

File: src/main/java/edu/ie3/datamodel/models/input/container/SystemParticipants.java [hunk 6]
```diff
@@ -110,6 +111,68 @@ public SystemParticipants(Collection<SystemParticipants> systemParticipants) {
...
+          "The provided entities in '"
+              + this.getClass().getSimpleName()
+              + "' contains duplicate UUIDs. "
+              + "This is not allowed!\nDuplicated uuids:\n\n"
+              + exceptionString);
+    }
+  }
+
   @Override
   public List<SystemParticipantInput> allEntitiesAsList() {
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/models/input/container/SystemParticipants.java:177]:
```suggestion
  public final List<SystemParticipantInput> allEntitiesAsList() {
```

Might as well resolve the SonarQube issue.

**Comment by johanneshiry** [comment]
@ckittl thanks for your comments - addressed all of them

sorry for decreased coverage but I think there is no sense in writing a test for the constructors only...

**Pull request merged**